### PR TITLE
Add performence tracer for nodes

### PR DESCRIPTION
**Pull request by shiyu22**
#2271

### CODE REVIEW

File: towhee/runtime/runtime_pipeline.py [hunk 1]
```diff
@@ -36,11 +40,13 @@ def __init__(self,
                  nodes: Dict[str, NodeRepr],
                  edges: Dict[str, Any],
                  operator_pool: 'OperatorPool',
-                 thread_pool: 'ThreadPoolExecutor'):
+                 thread_pool: 'ThreadPoolExecutor',
+                 time_profiler: 'TimeProfiler' = TimeProfiler()):
```
Comment (junjiejiangjjj) [inline towhee/runtime/runtime_pipeline.py:44]: Use None as the default value, and in `__init__`,  create TimeProfiler if the input is None.

File: towhee/runtime/performance_profiler.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+class TimeProfiler:
+    """
+    TimeProfiler to record the timestamp.
+    """
+    def __init__(self):
+        self._enable = False
+        self.time_record = []
+
+    def record(self, name, event, pipe_input=None):
```
Comment (junjiejiangjjj) [inline towhee/runtime/performance_profiler.py:232]: input & output nodes also need tracer.

Comment (shiyu22) [reply towhee/runtime/performance_profiler.py:232]: Get it, and do we need to show the performance for them?

Comment (junjiejiangjjj) [reply towhee/runtime/performance_profiler.py:232]: yeah

**Pull request merged**